src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -142,6 +142,30 @@ internal class AnalyzerManager
         }
 
         /// <summary>
+        /// Returns true if analyzer registered a compilation start action during <see cref="DiagnosticAnalyzer.Initialize(AnalysisContext)"/>
+        /// which registered a compilation end action and at least one other analyzer action, that the end action depends upon.
+        /// </summary>
+        public async Task<bool> GetAnalyzerHasDependentCompilationEndAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
```

Comment:
I *think* in the eventual world we need either of the following:

1. Have no IDE driver at all and IDE diagnostic service completely depend on compiler driver for analysis

2. If (1) above doesn't meet our perf requirements, and we are forced to have a separate IDE driver based on better public APIs from compiler's analyzer driver, we still don't want IDE driver to handle running compilation start actions. In fact, given we all agree AnalyzerActions won't be public, IDE driver would not be able run selective compilation start actions selectively.